### ISSUE TITLE
SC2039 Does not trigger on sh script

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#2003]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2039
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# apt-whatprovides ver. 201801010101 Copyright alexx, MIT Licence
# rdfa:deps="[realpath,apt-file,grep,which,sh,echo]"
# https://askubuntu.com/a/1057124/645458
#
#	Becautified
#
# Author: "Ntentos, Stavros" <stdedos@gmail.com>

BINARY="$(realpath "$(which "$@")" 2>/dev/null)"
[ -z "$BINARY" ] && BINARY="$@"

echo "Searching for ${BINARY}"
PACKAGE="$(apt-file search "${BINARY}" |grep -E ":.*[^-.a-zA-Z0-9]${BINARY}$")"

echo "${PACKAGE}"

```

#### Here's what shellcheck currently says:

```console
$ shellcheck apt-whatprovides.sh

In apt-whatprovides.sh line 10:
[ -z "$BINARY" ] && BINARY="$@"
                           ^--^ SC2124: Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.


For more information:
  https://www.shellcheck.net/wiki/SC2124 -- Assigning an array to a string! A...
```

#### Here's what I wanted or expected to see:

```console
$ shellcheck apt-whatprovides.sh

In apt-whatprovides.sh line 10:
[ -z "$BINARY" ] && BINARY="$@"
                           ^--^ SC2124: Assigning an array to a string! Assign as array, or use * instead of @ to concatenate.
                           ^--^ SC2039: In POSIX sh, arrays are undefined.


For more information:
  https://www.shellcheck.net/wiki/SC2124 -- Assigning an array to a string! A...
  https://www.shellcheck.net/wiki/SC2039 -- In POSIX sh, arrays are undefined.
```

**Comment by koalaman** [comment]
This code doesn't actually use arrays, and would work on sh (apart from the SC2124 problems). Maybe the warning should have said "an array-like expression" instead.

**Comment by stdedos** [comment]
Up to you - I trust your judgement.
